src/App.js

```diff
@@ -1,24 +1,36 @@
 import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { reference } from './lib/firebase';
+import { addDoc, getDocs } from 'firebase/firestore';
+import { useState } from 'react';
+import { useEffect } from 'react';
 
 function App() {
+  const [datas, setData] = useState([]);
+  const [newItems, setNewItems] = useState('Item0');
+  const [newItemsID, setNewItemsID] = useState(1);
+
+  const addItem = () => {
+    setNewItems('item' + newItemsID);
+    addDoc(reference, { Item: newItems, id: newItemsID });
+    setNewItemsID((prev) => prev + 1);
+  };
+
+  useEffect(() => {
+    getDocs(reference).then((snapshot) => {
+      setData(snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id })));
+    });
+  }, []);
+
+  const dataElements = datas.map((data) => (
+    <ul key={data.id}>
+      <li> {data.Item} </li>
+    </ul>
+  ));
+
```

Comment:
In addition to all the other comments, the reason how its not fetching immediately is because not event is set up to listen for changes in firestore. So onSnapShot will solve that as Reda pointed out!👍

Reply:
@OloyedeTobi are you still making changes to this PR? did you look at the other comment?🤔